desktop/src/gui/controller.rs

```diff
@@ -361,6 +338,52 @@ impl GuiController {
     }
 }
 
+fn create_wgpu_instance(
+    preferred_backends: wgpu::Backends,
+) -> anyhow::Result<(wgpu::Instance, wgpu::Backends)> {
+    for backend in preferred_backends.iter() {
+        if let Some(instance) = try_wgpu_backend(backend) {
+            tracing::info!(
+                "Using preferred backend {}",
+                format_list(&get_backend_names(backend), "and")
+            );
+            return Ok((instance, backend));
+        }
+    }
+
+    tracing::warn!(
+        "Preferred backend(s) of {} not available; falling back to any",
+        format_list(&get_backend_names(preferred_backends), "or")
+    );
+
+    for backend in wgpu::Backends::all() {
```

Comment:
this is a `bitflags`, so we could subtract the preferred ones to not try them twice.